src/test/java/seedu/address/logic/commands/UpdatePhotoCommandTest.java

```diff
@@ -0,0 +1,173 @@
+        } catch (CommandException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_invalidIndexInEmptyAddressBook_exceptionThrown() {
+        AddressBookParser parser = new AddressBookParser();
+        try {
+            Command command = parser.parseCommand("updatephoto 0 path/images/default_photo.png");
+            assertTrue(command instanceof UpdatePhotoCommand);
+            UpdatePhotoCommand updatePhotoCommand = (UpdatePhotoCommand) command;
+            updatePhotoCommand.execute(new ModelManager());
+        } catch (ParseException e) {
+            fail();
+        } catch (CommandException e) {
+            assertEquals(e.getMessage(), "The person index provided is invalid");
+        }
+    }
+
+    @Test
+    public void execute_nonPositiveIndexInNonEmptyAddressBook_exceptionThrown() {
+        AddressBookParser parser = new AddressBookParser();
+        try {
+            Command aCommand = parser.parseCommand(
+                    "add n/John Doe p/98765432 e/johndoe@gmail.com a/John street, block 123, #01-01 t/friend"
+            );
+            Command upCommand = parser.parseCommand("updatephoto 0 path/images/default_photo.png");
+            assertTrue(aCommand instanceof AddCommand);
+            assertTrue(upCommand instanceof UpdatePhotoCommand);
+            AddCommand addCommand = (AddCommand) aCommand;
+            UpdatePhotoCommand updatePhotoCommand = (UpdatePhotoCommand) upCommand;
+            ModelManager temp = new ModelManager();
+            addCommand.execute(temp);
+            updatePhotoCommand.execute(temp);
+        } catch (ParseException e) {
+            fail();
+        } catch (CommandException e) {
+            assertEquals(e.getMessage(), "The person index provided is invalid");
+        }
+    }
+
+    @Test
+    public void execute_outOfBoundIndexInNonEmptyAddressBook_exceptionThrown() {
+        AddressBookParser parser = new AddressBookParser();
+        try {
+            Command aCommand = parser.parseCommand(
+                    "add n/John Doe p/98765432 e/johndoe@gmail.com a/John street, block 123, #01-01 t/friend"
+            );
+            Command upCommand = parser.parseCommand("updatephoto 2 path/images/default_photo.png");
+            assertTrue(aCommand instanceof AddCommand);
+            assertTrue(upCommand instanceof UpdatePhotoCommand);
+            AddCommand addCommand = (AddCommand) aCommand;
+            UpdatePhotoCommand updatePhotoCommand = (UpdatePhotoCommand) upCommand;
+            ModelManager temp = new ModelManager();
+            addCommand.execute(temp);
+            updatePhotoCommand.execute(temp);
+        } catch (ParseException e) {
+            fail();
+        } catch (CommandException e) {
+            assertEquals(e.getMessage(), "The person index provided is invalid");
+        }
+    }
+
+    @Test
+    public void execute_invalidPath_exceptionThrown() {
```

Comment:
Just a suggestion, testing can be afford to be more rigourous, using the idea of [equivalence partitioning](https://nus-cs2103-ay2324s1.github.io/website/schedule/week10/topics.html#w10-4-test-cases-equivalence-partitioning), e.g. consider that `invalidPath` can be an empty string as well